### PR TITLE
[WNMGDS-2673] Audit and update default props

### PR DESCRIPTION
**Pull request by zarahzachz**
- Moved inline default prop values to `defaultProps` to ensure props table populates with something
- Updated Storybook props table

### CODE REVIEW

File: packages/design-system/src/components/VerticalNav/VerticalNavItemLabel.tsx [hunk 1]
```diff
@@ -55,8 +55,8 @@ export const VerticalNavItemLabel = (props: VerticalNavItemLabelProps): React.Re
 
   let otherProps;
   if (LabelComponent === 'button') {
-    const collapsedLabel = props.ariaCollapsedStateButtonLabel ?? t('verticalNav.expand');
-    const expandedLabel = props.ariaExpandedStateButtonLabel ?? t('verticalNav.collapse');
+    const collapsedLabel = props.ariaCollapsedStateButtonLabel;
+    const expandedLabel = props.ariaExpandedStateButtonLabel;
```
Comment (pwolfert) [inline packages/design-system/src/components/VerticalNav/VerticalNavItemLabel.tsx:59]: These could be inlined in the one place they're used now

**Review by pwolfert — COMMENTED** [review]
All these changes look correct, but I just found a problem that I don't have an immediate solution for: Unfortunately by evaluating these `t` function calls when the module loads, we can no longer update the language dynamically and expect the default content to change languages.

File: packages/design-system/src/components/VerticalNav/VerticalNavItemLabel.tsx (hunk 1 above)
Comment (zarahzachz) [reply packages/design-system/src/components/VerticalNav/VerticalNavItemLabel.tsx:59]: i just realized this is unnecessary to include - this sets the `title` attribute on a button and titles aren't accessible so this does nothing and should be removed

File: packages/design-system/src/components/Spinner/Spinner.tsx [hunk 2]
```diff
@@ -47,14 +47,15 @@ export const Spinner: React.FunctionComponent<SpinnerProps> = (props: SpinnerPro
 
   return (
     <span className={className} role={props.role}>
-      <span className="ds-u-visibility--screen-reader">{props['aria-valuetext']}</span>
+      <span className="ds-u-visibility--screen-reader">
+        {props['aria-valuetext'] ?? t('spinner.ariaText')}
```
Comment (zarahzachz) [inline packages/design-system/src/components/Spinner/Spinner.tsx:51]: reverting the spinner update - dynamically updating prop isn't supported when defined as defaultProp

File: packages/design-system/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx [hunk 3]
```diff
@@ -66,8 +66,4 @@ const ThirdPartyExternalLink = ({
   );
 };
 
-ThirdPartyExternalLink.defaultProps = {
-  learnMoreUrl: '',
```
Comment (zarahzachz) [inline packages/design-system/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx:70]: i don't know why this was an empty string but it seems like if we were to define a default value, it should be useful

File: packages/design-system/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx [hunk 4]
```diff
@@ -12,7 +12,7 @@ interface ThirdPartyExternalLinkProps {
   children: string;
   /** Additional classes to be applied to the external link button. */
   className?: string;
-  /** Specify the URL User's should visit to learn more about your application's external link policy. */
+  /** Specify the URL users should visit to learn more about your application's external link policy. */
   learnMoreUrl?: string;
```
Comment (zarahzachz) [inline packages/design-system/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx:16]: should this be a required prop?

**Review by pwolfert — APPROVED** [review]
LGTM 👍

**Pull request merged**